features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Security/AddPolicy.jsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import React, {Component} from 'react'
+
+import {Link} from 'react-router-dom'
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';;
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import MenuIcon from '@material-ui/icons/Menu';
+import Typography from '@material-ui/core/Typography';
+import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import { withStyles } from '@material-ui/core/styles';
+
+import API from 'AppData/api'
+import Alert from 'AppComponents/Shared/Alert'
+
+const styles = theme => ({
+    addNewWrapper: {
+        backgroundColor: theme.palette.background.paper,
+        color: theme.palette.getContrastText(theme.palette.background.paper),
+        border: 'solid 1px ' + theme.palette.grey['300'],
+        borderRadius: theme.shape.borderRadius,
+        marginTop: theme.spacing.unit*2,
+    },
+    addNewHeader: {
+        padding: theme.spacing.unit*2,
+        backgroundColor: theme.palette.grey['300'],
+        fontSize: theme.typography.h6.fontSize,
+        color: theme.typography.h6.color,
+        fontWeight: theme.typography.h6.fontWeight,
+    },
+    addNewOther: {
+        padding: theme.spacing.unit*2,
+    },
+    button: {
+        marginLeft: theme.spacing.unit*2,
+        textTransform: theme.custom.leftMenuTextStyle,
+        color: theme.palette.getContrastText(theme.palette.primary.main),
+    },
+    contentWrapper: {
+        maxWidth: theme.custom.contentAreaWidth,
+    },
+    addJsonContent: {
+        whiteSpace: 'pre',
+    }
+});
+
+class AddPolicy extends Component {
+    state = {
+        selectedPolicy : {uuid: '', name: 'Select', policy: '', type: ''},
+        policies: []
+    };
+
+    constructor(props) {
+        super(props);
+        let api = new API();
+        let promisedApi = api.get(this.props.id);
+        promisedApi.then(response => {
```

Comment:
Please don't call `this.setState` in Component constructor,Please refer the [Docs](https://reactjs.org/docs/react-component.html#constructor)

> You should not call setState() in the constructor(). Instead, if your component needs to use local state, assign the initial state to this.state directly in the constructor

Reply:
Moved api call and this.setState call to componentDidMount().